### DISCUSSION TITLE
Ethernet and aggressive buffering reports error 22 or 33 and locks up

### DISCUSSION THREAD

**Discussion by Meliusja** [terjeio/grblHAL#339] [Q&A]
I was hoping the new teensyduino would fix my Ethernet problems but they seem to still be there.  First off I am using Teensyduino 1.55 because I can't figure out how or where to download 1.54.   I am running a version of grblhal from 12/5/2021 with the only build option changes from default being Ethernet and backlash comp enabled.   It errors almost immediately in check mode on the attached file.   My machine is 3 axis with speed 5000 and acc 500.  I see Phil Barrett had a similar thread which was closed due to teensyduino 1.54.   It runs to completion with aggressive buffering turned off.  Any help will be greatly appreciated.  I'm more than happy to provide anything else like setting files exact versions etc.

Update:
I found Teensyduino 1.54 and tried it but it does not help.

Thanks
Jeff Melius

[1 FrameBack 6mm Flat 3F.zip](https://github.com/terjeio/grblHAL/files/7699380/1.FrameBack.6mm.Flat.3F.zip)

**Comment by phil-barrett** [comment]
With a build from July (20210715) and built with TD 1.54, I get several different errors running when connected via ethernet.  I do not get those errors when connected via USB.  The screens shots are with the XL version of IO sender but same error with ioSender 2.0.35. Also, when terminating ioSender, I get an exception (shown below).   I do not get the error with ioSender 2.0.35. (will report on ioSender github repo. Jeff, if you don't mind, I will submit your gcode file with it as well).
![image](https://user-images.githubusercontent.com/16570958/145729402-fcb3f92c-43e6-4c9b-9c25-7e52fb82e189.png)
![image](https://user-images.githubusercontent.com/16570958/145729477-6fd08b2b-c8e9-4e0d-bfbe-91732a78de3c.png)

**Reply by terjeio** [reply]
There is a bug somewhere when connecting via telnet, via websocket it works. The STM32F756 driver does not fail indicating this is a Teensy issue, perhaps in the network stack as the telnet code is exactly the same for both.

**Reply by Meliusja** [reply]
I just tried websockets and they work great and fast with buffering enabled.   This solution works fine for me.   
Thanks,
Jeff

**Reply by Meliusja** [reply]
Phil,
Feel free to attach Gcode to sender issue.
Jeff

**Comment by terjeio** [comment]
I have successfully made a workaround for this issue but I guess I have to dig deeper into lwIP programming since the other drivers do not show this problem. I will commit the fix soon.

Phil, when you have time can you change the links you have to the new repo - I am planning to archive this.

**Reply by terjeio** [reply]
Workaround now committet in the [new repo](https://github.com/grblHAL/iMXRT1062).

**Reply by Meliusja** [reply]
I'm just curious,  is the work around the change to slightly slow polling rate?
Thanks,
Jeff

**Reply by terjeio** [reply]
No, it is mainly about [polling when data arrives](https://github.com/grblHAL/Plugin_networking/blob/45ebf09cc46f555b115a9f7bc327a389860770bb/TCPStream.c#L356-L358). What is strange is that other drivers did not play well with this change.
